.github/workflows/build.yml

```diff
@@ -0,0 +1,125 @@
+name: R Docker
+
+on:
+  push:
+  workflow_dispatch:
+    inputs:
+      variants:
+        description: |
+          Space-separated list of variants (platforms). Defaults to all variants.
+        required: false
+        default: ''
+        type: string
+      versions:
+        description: |
+          Space-separated list of R versions. Defaults to all versions.
+        required: false
+        default: ''
+        type: string
+      publish_images:
+        description: |
+          Publish images to Docker Hub?
+        required: false
+        default: false
+        type: boolean
+      include_patch_versions:
+        description: |
+          When building all R versions, include R patch version tags (x.y.z) as well?
+        required: false
+        default: false
+        type: boolean
+  workflow_call:
+    inputs:
+      versions:
+        description: |
+          Space-separated list of R versions. Defaults to all versions.
+        required: false
+        default: ''
+        type: string
+      publish_images:
+        description: |
+          Publish images to Docker Hub?
+        required: false
+        default: false
+        type: boolean
+      include_patch_versions:
+        description: |
+          Include R patch versions?
+        required: false
+        default: false
+        type: boolean
+    secrets:
+      DOCKERHUB_USERNAME:
+        required: true
+      DOCKERHUB_TOKEN:
+        required: true
```

Comment:
Why are these required if publishing can be skipped?

Reply:
There's a Docker login step that always gets run in the main workflow. In hindsight, I guess we could skip that step if publishing is skipped, but the secrets will be present regardless, and it is kind of useful to test that valid credentials are configured without having to publish an image.

Also wondering now, passing secrets may not even be necessary if the reusable workflow is in the same repo. I'll test if this config is even necessary and remove it if not.

Reply:
It's apparently required even for reusable actions in the same repo, so leaving this in.